exe/Main.hs

```diff
@@ -147,15 +160,35 @@ newSession' cradle = do
     initDynLinker env
     newHscEnvEq env
 
-loadEnvironment :: FilePath -> IO (FilePath -> Action HscEnvEq)
-loadEnvironment dir = do
-    res <- liftIO $ newSession' =<< getCradle dir
-    return $ const $ return res
-
-getCradle :: FilePath -> IO Cradle
-getCradle dir = do
-    dir <- pure $ addTrailingPathSeparator dir
-    mbYaml <- findCradle dir
-    case mbYaml of
-        Nothing -> loadImplicitCradle dir
-        Just yaml -> loadCradle yaml
+
+loadSession :: FilePath -> IO (FilePath -> Action HscEnvEq)
+loadSession dir = do
+    cradleLoc <- memoIO $ \v -> do
+        res <- findCradle v
+        -- Sometimes we get C: and sometimes we get c:, try and normalise that
+        -- e.g. see https://github.com/digital-asset/ghcide/issues/126
+        case res of
+            Nothing -> return Nothing
+            Just res -> Just <$> canonicalizePath res
```

Comment:
Do we really want to resolve symlinks? It isn’t clear to me that this is the right thing to do here. If you just want to normalise drive letters `normalise` should do the job.

Reply:
Yep, that seems sufficient. Although once we normalise drive letters, we can probably avoid it entirely.